datalad/distribution/subdatasets.py

```diff
@@ -56,67 +66,68 @@
 valid_key = re.compile(r'^[A-Za-z][-A-Za-z0-9]*$')
 
 
-def _parse_gitmodules(dspath):
-    gitmodule_path = opj(dspath, ".gitmodules")
-    parser = GitConfigParser(gitmodule_path)
+def _parse_gitmodules(ds):
+    gitmodules = ds.pathobj / '.gitmodules'
+    if not gitmodules.exists():
+        return {}
+    # pull out file content
+    out, err = ds.repo._git_custom_command(
+        '',
+        ['git', 'config', '-z', '-l', '--file', '.gitmodules'])
+    # abuse our config parser
+    db, _ = _parse_gitconfig_dump(out, {}, None, True)
     mods = {}
-    for sec in parser.sections():
-        try:
-            modpath = parser.get(sec, 'path')
-        except Exception:
-            lgr.debug("Failed to get '%s.path', skipping section", sec)
+    for k, v in iteritems(db):
+        if not k.startswith('submodule.'):
+            # we don't know what this is
```

Comment:
I think it would be useful to leave some debug message at least if something unexpected discovered.  could as well be a warning since AFAIK and according to `man gitmodules` there should be no other sections there.

Reply:
Done